### ISSUE TITLE
Competing consumers pattern doesn't work in JetStream

### ISSUE THREAD

**Issue by aserricchio** [dapr/components-contrib#2827]
## Expected Behavior

With JetStream in a Deployment in Kubernetes each message should be processed by only one pod with the same app-id

## Actual Behavior

The message is processed by all the pods.

## Steps to Reproduce the Problem

I've used this [tutorial](https://github.com/dapr/quickstarts/tree/master/tutorials/pub-sub#run-in-kubernetes) replacing redis with jetstream, using the following yaml:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: pubsub
spec:
  type: pubsub.jetstream
  version: v1
  metadata:
  - name: natsURL
    value: "nats://dapr-nats.default.svc.cluster.local:4222"
```

The tutorial is working as expected using Redis.
I was also wondering if consumerID is supported by JetStream, I can't find info in the documentation but the same parameter is present in NATS Streaming module. The test with Redis and the StatefulSet worked as [expected](https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-subscribe-statefulset/).

## Release Note

RELEASE NOTE:

**Comment by berndverst** [comment]
Whether the competing consumer pattern works depends on the component implementation and the capabilities of the service.

Note that NATS Streaming is a completely separate component and implementation (made by different people) - so there is no point in comparing these components.

`consumerID` is not supported - I tried adding this before, which broke some Jetstream behavior :)

What you can try however @aserricchio is using `"{podName}"` as the value for one of these metadata options:

```
  - name: name
    value: "my-conn-name"
  - name: streamName
    value: "my-stream"
  - name: durableName 
    value: "my-durable"
  - name: queueGroupName
    value: "my-queue"
```

I don't know anything about Jetstream specifically, so I don't know what these concepts are. But you can try using the dynamic podname value instead.

https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-jetstream/

I'll leave it up to @Jarema who is one of the maintainers from NATS to investigate this.

At the moment there is no reason to believe there is a bug here - you probably just need to configure the dynamic value for the appropriate property.

**Comment by Jarema** [comment]
In NATS, a Consumer can be treated as a view into a Stream.

Each consumer tracks its progress, and unless it's a work queue Stream, one pull consumer, or push consumer with queue group, each consumer will all/specified messages from that stream independently. 

To achieve your goal, the easiest way is to create a consumer with a queue group name, i.e.:

```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: pubsub
spec:
  type: pubsub.jetstream
  version: v1
  metadata:
  - name: natsURL
    value: "nats://dapr-nats.default.svc.cluster.local:4222"
  - name: durableName 
    value: "my-durable"
  - name: queueGroupName
    value: "my-queue"
```

**Comment by msfussell** [comment]
Would be great if someone could pick up this docs issue https://github.com/dapr/docs/issues/3382

**Comment by aserricchio** [comment]
Hi @Jarema I've tried using `queueGroupName` but I think it's something missing, I keep getting this error inside the subscribers:
> cannot create a queue subscription for a consumer without a deliver group

I'm using the following versions:
- Dapr: 1.10.5
- Nats: 2.9.16

**Comment by bruth** [comment]
☝️ Indeed, I just checked and it is missing from being set on the consumer config: https://github.com/dapr/components-contrib/blob/5118b73cfc942ada8f25824ae194fac0805c4a15/pubsub/jetstream/jetstream.go#L144

Opened a PR to fix.

**Comment by berndverst** [comment]
This will be fixed in the upcoming Dapr 1.11 release - in 1-2 weeks.
